include/pybind11/pybind11.h

```diff
@@ -2121,7 +2121,12 @@ class gil_scoped_acquire {
                     pybind11_fail("scoped_acquire::dec_ref(): internal error!");
             #endif
             PyThreadState_Clear(tstate);
+#if (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION > 6) && !defined(Py_LIMITED_API)
```

Comment:
No point in checking `Py_LIMITED_API`, since pybind11 has never cared about that, but does pypy have the same problem, or does it behave better?

Also, `#if PY_VERSION_HEX >= 0x03060000` seems like a better check.